src/main/java/com/squareup/kotlinpoet/KotlinFile.kt

```diff
@@ -194,6 +202,24 @@ class KotlinFile private constructor(builder: KotlinFile.Builder) {
       require(SourceVersion.isName(fileName)) { "not a valid file name: $fileName" }
     }
 
+    fun addFileAnnotation(annotationSpec: AnnotationSpec) = apply {
+      fileAnnotations += annotationSpec.ensureFileUseSiteTarget()
+    }
+
+    private fun AnnotationSpec.ensureFileUseSiteTarget() = when (useSiteTarget) {
```

Comment:
This is only used in one place so I think we can just inline it into the `addFileAnnotation(AnnotationSpec)` method.

Also I would throw an exception if any other use-site target is already specified (with a test for that).